### PR TITLE
TICKscript Editor

### PR DESCRIPTION
**Pull request by 121watts**
- [x] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [x] Rebased/mergable
  - [x] Tests pass
  - [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

Connect #789 

### The problem
Sometimes you just want to write your own tick script.

### The Solution
Provide a text editor for the user to do so.

### CODE REVIEW

**Comment by alexpaxton** [comment]
WIP screenshot

screen shot 2017-07-25 at 5 31 19 pm

**Comment by lukevmorris** [comment]
Blocked on #1733

**Comment by cryptoquick** [comment]
First, I'm glad we can finally get this in, so great work, to everyone involved
I double-checked your refactor of MultiSelectDropdown, it should be good
I'm not sure if this is a regression, but saving a rule if there is an error now produces a copy of that rule, not sure if this is good or desired behavior...
More to come, this is a very large PR, please be patient as I review it.

File: ui/src/kapacitor/actions/view/index.js [hunk 1]
```diff
@@ -208,3 +235,46 @@ export function updateRuleStatus(rule, status) {
...
+  sourceID
+) => async dispatch => {
+  try {
+    const {data} = await createTaskAJAX(kapacitor, task)
+    router.push(`/sources/${sourceID}/alert-rules`)
+    dispatch(publishNotification('success', 'You made a TICKscript!'))
+    return data
+  } catch (error) {
+    if (!error) {
+      dispatch(errorThrown('Could not communicate with server'))
```
[the remark this answers is not in the document]
Comment (121watts) [reply ui/src/kapacitor/actions/view/index.js]: I agree 💯

[the remark this answers is not in the document]
Comment (cryptoquick) [reply ui/src/kapacitor/actions/view/index.js]: Let's do this in an another issue, like #1539 . :shipit:

File: ui/src/admin/components/UserRow.js [hunk 2]
```diff
@@ -104,8 +103,8 @@ const UserRow = ({
       <td>
         {allPermissions && allPermissions.length
           ? <MultiSelectDropdown
-              items={allPermissions}
-              selectedItems={_.get(permissions, ['0', 'allowed'], [])}
+              items={allPermissions.map(p => ({name: p}))}
+              selectedItems={perms.map(p => ({name: p}))}
```
[the remark this answers is not in the document]
Comment (cryptoquick) [reply ui/src/admin/components/UserRow.js]: This has been fixed, but other issues with the backend have come up. See #2021

**Comment by cryptoquick** [comment]
Good stuff, reviewed and ready for merge.

**Pull request merged**